patches/0006-Vendor-crypto-backends.patch

```diff
@@ -6854,6 +7321,147 @@ index 00000000000000..5de62f95a7a4ec
 +	}
 +	return nil
 +}
++
++// fetchTLS1PRF3 fetches the TLS1-PRF KDF algorithm.
++// It is safe to call this function concurrently.
++// The returned EVP_KDF_PTR shouldn't be freed.
++var fetchTLS1PRF3 = sync.OnceValues(func() (C.GO_EVP_KDF_PTR, error) {
++	checkMajorVersion(3)
++
++	name := C.CString("TLS1-PRF")
++	kdf := C.go_openssl_EVP_KDF_fetch(nil, name, nil)
++	C.free(unsafe.Pointer(name))
++	if kdf == nil {
++		return nil, newOpenSSLError("EVP_KDF_fetch")
++	}
++	return kdf, nil
++})
++
++// tls1PRF3 implements TLS1PRF for OpenSSL 3 using the EVP_KDF API.
++func tls1PRF3(result, secret, label, seed []byte, md C.GO_EVP_MD_PTR) error {
++	checkMajorVersion(3)
++
++	kdf, err := fetchTLS1PRF3()
++	if err != nil {
++		return err
++	}
++	ctx := C.go_openssl_EVP_KDF_CTX_new(kdf)
++	if ctx == nil {
++		return newOpenSSLError("EVP_KDF_CTX_new")
++	}
++	defer C.go_openssl_EVP_KDF_CTX_free(ctx)
++
++	bld, err := newParamBuilder()
++	if err != nil {
++		return err
++	}
++	bld.addUTF8String(_OSSL_KDF_PARAM_DIGEST, C.go_openssl_EVP_MD_get0_name(md), 0)
++	bld.addOctetString(_OSSL_KDF_PARAM_SECRET, secret)
++	bld.addOctetString(_OSSL_KDF_PARAM_SEED, label)
++	bld.addOctetString(_OSSL_KDF_PARAM_SEED, seed)
++	params, err := bld.build()
++	if err != nil {
++		return err
++	}
++	defer C.go_openssl_OSSL_PARAM_free(params)
++
++	if C.go_openssl_EVP_KDF_derive(ctx, base(result), C.size_t(len(result)), params) != 1 {
++		return newOpenSSLError("EVP_KDF_derive")
++	}
++	return nil
++}
+diff --git a/src/vendor/github.com/golang-fips/openssl/v2/zaes.go b/src/vendor/github.com/golang-fips/openssl/v2/zaes.go
+new file mode 100644
+index 00000000000000..4d945290e3bdff
+--- /dev/null
++++ b/src/vendor/github.com/golang-fips/openssl/v2/zaes.go
+@@ -0,0 +1,86 @@
++// Code generated by cmd/genaesmodes. DO NOT EDIT.
++
++//go:build !cmd_go_bootstrap
```

Comment:
Got it--it's because this doesn't include cgo:

```suggestion
+//go:build !cmd_go_bootstrap && cgo
```

Nearly every other file has `import "C"` which implicitly requires cgo.

Reply:
You saved my day! ❤️

Reply:
https://github.com/golang-fips/openssl/pull/209